app/scripts/nav.js

```diff
@@ -27,18 +29,29 @@ $(function() {
 
     lastScrollTop = currentScrollTop;
 
-    if (isOnTop(currentScrollTop) && !nav.hasClass(onTopClass) ||
-        (!isOnTop(currentScrollTop) && nav.hasClass(onTopClass)))
-      toggleNavOnTop();
+    if (lastClass === onTopClass || lastClass === onTopClassIndex) {
+      toggleNavOnTop(lastClass, currentScrollTop);
+    }
 
   }
 
   function isOnTop(currentScrollTop) {
     return currentScrollTop < delta;
   }
 
-  function toggleNavOnTop() {
-    nav.toggleClass(onTopClass);
+  function getLastClass() {
```

Comment:
I would rename this to `getTopClass`, since it has only to do with the top class, not necessarily the _last_ class.